spyder_line_profiler/spyder/plugin.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# Copyright © 2021, Spyder Line Profiler contributors
+#
+# Licensed under the terms of the MIT license
+# ----------------------------------------------------------------------------
+"""
+Spyder Line Profiler 5 Plugin.
+"""
+
+# Third-party imports
+from qtpy.QtGui import QIcon
+
+# Spyder imports
+from spyder.api.plugins import Plugins, SpyderDockablePlugin
+from spyder.api.translations import get_translation
+from spyder.api.plugin_registration.decorators import on_plugin_available
+
+# Local imports
+from spyder_line_profiler.spyder.confpage import SpyderLineProfilerConfigPage
+from spyder_line_profiler.spyder.widgets import SpyderLineProfilerWidget
+
+_ = get_translation("spyder_line_profiler.spyder")
+
+
+class SpyderLineProfiler(SpyderDockablePlugin):
+    """
+    Spyder Line Profiler plugin for Spyder 5.
+    """
+
+    NAME = "spyder_line_profiler"
+    REQUIRES = [Plugins.Editor]
+    OPTIONAL = []
+    TABIFY = Plugins.Help
+    WIDGET_CLASS = SpyderLineProfilerWidget
+    CONF_SECTION = NAME
+    CONF_WIDGET_CLASS = SpyderLineProfilerConfigPage
+
+    # --- Signals
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    @staticmethod
+    def get_name():
+        return _("Line Profiler")
+
+    def get_description(self):
+        return _("Line profiler display for Spyder 5")
+
+    def get_icon(self):
+        return QIcon('./data/images/spyder.line_profiler.png')
+
+
+    def on_initialize(self):
+        widget = self.get_widget()
+
+        # run_action = self.create_action(
+        #     ProfilerActions.ProfileCurrentFile,
+        #     text=_("Run profiler"),
+        #     tip=_("Run profiler"),
+        #     icon=self.create_icon('profiler'),
+        #     triggered=self.run_profiler,
+        #     register_shortcut=True,
+        # )
+
+        # run_action.setEnabled(is_profiler_installed())
+        # editor = self.get_plugin(Plugins.Editor)
+        # widget.sig_edit_goto_requested.connect(editor.load)
```

Comment:
I think we can remove this lines since they are already covered with the `on_editor_available` method